token.go

```diff
@@ -139,5 +143,9 @@ func retrieveToken(ctx context.Context, c *Config, v url.Values) (*Token, error)
 	if err != nil {
 		return nil, err
 	}
-	return tokenFromInternal(tk), nil
+	token := tokenFromInternal(tk)
+	if notifierFunc, ok := ctx.Value(TokenRefreshNotifier).(TokenRefreshNotifierFunc); ok {
+		notifierFunc(token)
+	}
```

Comment:
This is to propagate the new token back to the lib consumer whenever it gets refreshed ([example usage](https://github.com/flynn/flynn/blob/installer-azure/installer/azure_cluster.go#L22)).